### DISCUSSION TITLE
Migration from @walletconnect/web3-provider to @walletconnect/ethereum-provider

### DISCUSSION THREAD

**Discussion by desmowater** [WalletConnect/walletconnect-monorepo#2476] [Developer Support] [answered]
I am using @walletconnect/web3-provider for let user connect their wallet to my dapp,

- from PC browser without metamask extention, by scanning QR code with mobile metamask
- from mobile browser ex. safari, chorme (not metamask app browser)

and my Vue.js code is like this.
```
this.provider = new WalletConnectProvider({
  infuraId: this.config.INFURA_ID,
  qrcodeModalOptions: {
    desktopLinks: [],
    mobileLinks: ["metamask"],
  },
 });
await this.provider.enable();
this.web3Provider = new providers.Web3Provider(this.provider);
```
QR code looks like this.
![before](https://github.com/WalletConnect/walletconnect-monorepo/assets/110516827/cdcd0e59-e98d-41f8-a2a8-a61a0d9bb951)

Now trying to migrate to @walletconnect/ethereum-provider like this.
```
this.provider = await EthereumProvider.init({
  projectId: this.config.MY_PROJECT_ID,
  chains: [1],
  showQrModal: true,
  qrModalOptions: {
    desktopLinks: [],
    mobileLinks: ["metamask"],
  }, 
});
await this.provider.enable();
this.web3Provider = new providers.Web3Provider(this.provider);
```
So the QR code is,
![after](https://github.com/WalletConnect/walletconnect-monorepo/assets/110516827/51432155-fb5c-4297-b423-19fb6b2f4b3a)

and this can't be scanned by mobile metamask.

I suppose 'qrModalOptions' above is not appropriate, but can't find the answer so far.
I would be very grateful if a specific code could be provided.

**Comment by glitch-txs** [comment, marked as answer]
Hi! Your code looks good :) MetaMask currently doesn't support WalletConnect v2 but they will very soon, this is why it's currently not working

You won't need to add it in the qrModalOptions btw

**Reply by desmowater** [reply]
Thanks, @glitch-txs ! I will wait for Metamask support to start.
